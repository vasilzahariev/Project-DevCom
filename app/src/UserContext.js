import React from 'react';

const UserContext = React.createContext({
    user: null,
    login: (userObj) => {},
    logout: () => {}
});

export default UserContext;
