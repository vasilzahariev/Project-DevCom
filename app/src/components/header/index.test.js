import React from 'react';
import renderer from 'react-test-renderer';
import Header from './index';
import { BrowserRouter, Route } from 'react-router-dom';
import UserContext from '../../contexts/UserContext';

describe('Header', () => {
    it('should render HeaderUser', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: true
                        }
                    }}>
                        <Header />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
    it('should render HederUserGuest', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: false
                        }
                    }}>
                        <Header />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
});
