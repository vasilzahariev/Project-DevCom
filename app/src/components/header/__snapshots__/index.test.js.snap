// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render HeaderUser 1`] = `
<header
  className=""
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        className="MuiGrid-root container MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/news"
            onClick={[Function]}
          >
            News
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/forum"
            onClick={[Function]}
          >
            Forum
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/jobs"
            onClick={[Function]}
          >
            Jobs
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
          >
            <div
              className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
                htmlFor="custom-css-standard-input"
                id="custom-css-standard-input-label"
              >
                Search
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                onClick={[Function]}
              >
                <div
                  className="MuiInputAdornment-root MuiInputAdornment-positionStart"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <input
                  aria-invalid={false}
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                  disabled={false}
                  id="custom-css-standard-input"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              className="MuiFormControl-root WithStyles(ForwardRef(FormControl))-root-2"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="age-native-simple"
              >
                Category
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
                style={Object {}}
              >
                <select
                  aria-invalid={false}
                  className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
                  disabled={false}
                  multiple={false}
                  name="category"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  value="news"
                >
                  <option
                    value="news"
                  >
                    News
                  </option>
                  <option
                    value="forum"
                  >
                    Forum Posts
                  </option>
                  <option
                    value="jobs"
                  >
                    Jobs
                  </option>
                  <option
                    value="projects"
                  >
                    Projects
                  </option>
                  <option
                    value="users"
                  >
                    Users
                  </option>
                </select>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <input
              className="btn"
              disabled={true}
              type="submit"
              value="Search"
            />
          </div>
        </div>
      </form>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="badge"
            href="/chat"
            onClick={[Function]}
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "fontSize": 25,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
              >
                0
              </span>
            </span>
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div>
            <button
              aria-haspopup="true"
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiAvatar-fallback"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                    />
                  </svg>
                </div>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header should render HederUserGuest 1`] = `
<header
  className="header"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        className="MuiGrid-root container MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/news"
            onClick={[Function]}
          >
            News
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/forum"
            onClick={[Function]}
          >
            Forum
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="link"
            href="/jobs"
            onClick={[Function]}
          >
            Jobs
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
          >
            <div
              className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
                htmlFor="custom-css-standard-input"
                id="custom-css-standard-input-label"
              >
                Search
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                onClick={[Function]}
              >
                <div
                  className="MuiInputAdornment-root MuiInputAdornment-positionStart"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <input
                  aria-invalid={false}
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                  disabled={false}
                  id="custom-css-standard-input"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              className="MuiFormControl-root WithStyles(ForwardRef(FormControl))-root-2"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="age-native-simple"
              >
                Category
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
                style={Object {}}
              >
                <select
                  aria-invalid={false}
                  className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
                  disabled={false}
                  multiple={false}
                  name="category"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  value="news"
                >
                  <option
                    value="news"
                  >
                    News
                  </option>
                  <option
                    value="forum"
                  >
                    Forum Posts
                  </option>
                  <option
                    value="jobs"
                  >
                    Jobs
                  </option>
                  <option
                    value="projects"
                  >
                    Projects
                  </option>
                  <option
                    value="users"
                  >
                    Users
                  </option>
                </select>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <input
              className="btn"
              disabled={true}
              type="submit"
              value="Search"
            />
          </div>
        </div>
      </form>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        />
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <a
            className="link"
            href="/auth/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <a
            className="link"
            href="/auth/register"
            onClick={[Function]}
          >
            Register
          </a>
        </div>
      </div>
    </div>
  </div>
</header>
`;
